sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -472,6 +511,54 @@ void Scheduler::cleanupCommands(const std::vector<Command *> &Cmds) {
   }
 }
 
+void Scheduler::deferMemObjRelease(const std::shared_ptr<SYCLMemObjI> &MemObj) {
+  {
+    std::lock_guard<std::mutex> Lock{MDeferredMemReleaseMutex};
+    MDeferredMemObjRelease.push_back(MemObj);
+  }
+  cleanupDeferredMemObjects(false);
```

Comment:
```suggestion
  cleanupDeferredMemObjects(/*Blocking*/false);
```

Reply:
fixed in https://github.com/intel/llvm/pull/6837/commits/d4537a3c85672ebd6fe7ccee110b98c5d3f46b5b